src/io/aviso/taxi_toolkit/composite_assertions.clj

```diff
@@ -24,25 +24,25 @@
                                          (is true)
                                          (do (swap! j dec)
                                              (throw (ex-info "assertion has thrown an exception" {})))))]
-                  (is (retry-find+assert find-fn assert-fn))
+                  (is (retry-find+assert (System/currentTimeMillis) find-fn assert-fn))
                   (is (= 0 @i))
-                  (is (retry-find+assert find-fn throw-fn))
+                  (is (retry-find+assert (System/currentTimeMillis) find-fn throw-fn))
                   (is (= 0 @j))))}
-  [find-fn assert-fn]
+  [start find-fn assert-fn]
   (let [result (atom false)
         test-report test/report
         spy-report (fn [data]
                      (case (:type data)
-                       :fail nil
-                       :error nil
+                       (:fail :error) nil
                        :pass (do (reset! result true)
                                  (test-report data))
                        (test-report data)))
         retriable #(let [return (assert-fn (find-fn))]
-                    (when-not @result (throw (ex-info "forcing a retry" {})))
-                    return)
-        return (with-redefs [test/report spy-report]
-                 (try (retry-times retriable)
+                     (if return
+                       (reset! result true)
+                       (throw (ex-info "forcing a retry" {}))))
+        return (binding [test/report spy-report]
```

Comment:
Yeah, test2junit itself uses binding. I now developed a habit of running `lein test2junit :only blahblah` locally as well.